core/src/kura.rs

```diff
@@ -77,7 +78,7 @@ impl Kura {
             mode,
             block_store: Mutex::new(Box::new(block_store)),
             block_hash_array: Mutex::new(Vec::new()),
-            // broker,
+            block_arc_array: Mutex::new(Vec::new()),
```

Comment:
Consider `blocks` and `block_hashes`, given that the implementation detail of the backing storage is irrelevant.

Reply:
`block_arcs` and `block_hashes` then, because the `Arc` bit is significant.